code/modules/modular_computers/file_system/programs/statusdisplay.dm

```diff
@@ -15,36 +15,76 @@
 	var/upper_text
 	var/lower_text
 
-/datum/computer_file/program/status/proc/SendSignal()
+/**
+ * Post status display radio packet.
+ * Arguments:
+ * * command - the status display command
+ * * data1 - the data1 value, as defined by status displays
+ * * data2 - the data2 value, as defined by status displays
+ */
+/datum/computer_file/program/status/proc/post_status(command, data1, data2)
 	var/datum/radio_frequency/frequency = SSradio.return_frequency(FREQ_STATUS_DISPLAYS)
-
 	if(!frequency)
 		return
 
-	var/datum/signal/status_signal = new(list("command" = "message"))
+	var/datum/signal/status_signal = new(list("command" = command))
+	switch(command)
+		if("message")
+			status_signal.data["msg1"] = data1
+			status_signal.data["msg2"] = data2
+		if("alert")
+			status_signal.data["picture_state"] = data1
+
+	frequency.post_signal(src, status_signal)
 
-	status_signal.data["msg1"] = reject_bad_text(upper_text || "", MAX_STATUS_LINE_LENGTH)
-	status_signal.data["msg2"] = reject_bad_text(lower_text || "", MAX_STATUS_LINE_LENGTH)
+/**
+ * Post a message to status displays
+ * Arguments:
+ * * upper - Top text
+ * * lower - Bottom text
+ */
+/datum/computer_file/program/status/proc/post_message(upper, lower)
+	post_status("message", upper, lower)
 
-	frequency.post_signal(computer, status_signal)
+/**
+ * Post a picture to status displays
+ * Arguments:
+ * * picture - The picture name
+ */
+/datum/computer_file/program/status/proc/post_picture(picture)
+	var/static/obj/machinery/computer/communications/comm_prototype = /obj/machinery/computer/communications
 
-/datum/computer_file/program/status/proc/SetText(position, text)
-	switch(position)
-		if("upper")
-			upper_text = text
-		if("lower")
-			lower_text = text
+	if (!(picture in initial(comm_prototype.approved_status_pictures)))
+		return
+	if(picture in initial(comm_prototype.state_status_pictures))
+		post_status(picture)
+	else
+		post_status("alert", picture)
 
 /datum/computer_file/program/status/ui_act(action, list/params, datum/tgui/ui)
 	. = ..()
 	if(.)
 		return
 
 	switch(action)
-		if("stat_send")
-			SendSignal()
+		if("stat_message")
+			post_message(upper_text, lower_text)
+		if("stat_picture")
+			post_picture(params["picture"])
 		if("stat_update")
-			SetText(params["position"], params["text"]) // i hate the player i hate the player
+			var/text = reject_bad_text(params["text"] || "", MAX_STATUS_LINE_LENGTH)
+
+			switch(params["position"])
+				if("upper")
+					upper_text = text
+				if("lower")
+					lower_text = text
+
+/datum/computer_file/program/status/ui_static_data(mob/user)
+	var/list/data = list()
+	data["maxStatusLineLength"] = MAX_STATUS_LINE_LENGTH
```

Comment:
Wow, I could have swore that was possible.  Maybe I was getting it mixed up with Lua syntax.  Regardless I amended my comments to remove mention of using dot syntax since it was incorrect.  

Thanks for pointing it out!

Reply:
What is this in regards to?

Reply:
I had earlier suggested the wrong syntax in my review. Stylemistake point out it would lead to errors.